### DISCUSSION TITLE
Sourcing a script

### DISCUSSION THREAD

**Discussion by ISchwandner** [IBAHCM/RPiR#476] [Archive] [answered]
Hi Richard and Sonia,

I have created an R-script that creates the abundance data, so that I can source it at the top of my demos, and it works fine, but when I try to compile the reports R can't find it. I've tried putting it in different subfolders in the project or just leaving it in the main project folder and none of it seems to work. I've tried changing the source command to
source("myfile.R")
source("./myfile.R")
source("Myfolder/myfile.R") and nothing seemed to help with compiling, so now I have set it to the absolute file path:
source(C:/User/Git/Myproject/Myfile.R") and that works for compiling but I realise that this will not compile on anyone else's machine without being changed.
Is there a specific location within my project or a specific sub folder I should create to store this file for this to work? Or can I do something along the lines of getwd() and setwd() to ensure it runs on other machines?

While I'm here I also have a question about Project 3 and the comment in the report. Am I understanding correctly that we are to comment merely on the difference in values for one population as q increases rather than comparing the same diversity measures between different populations?

Many thanks!
Imogen

**Comment by richardreeve** [comment, marked as answer]
@ISchwandner - this has been tricky in the past, and I was about to suggest a slightly complicated way of dealing with it involving `devtools::wd()`, but I see that in classic R fashion there is now a package called `here` that seems to deal it automatically. It seems to be part of R's standard library, so it should be installed already, but you can find documentation here - https://github.com/r-lib/here - and a bit more on it (slightly out of date) here - https://github.com/jennybc/here_here - along with the immortal line:

> If the first line of your #rstats script is setwd("C:\Users\jenny\path\that\only\I\have"), I will come into your lab and SET YOUR COMPUTER ON FIRE.

Let me know if those two pages don't explain things well enough between them...

In answer to your second question, yes. How does diversity change as q increases?

**Reply by ISchwandner** [reply]
Hi Richard,

Hahahah I will make sure that's not the case. Thanks a lot for that. I will try it.

For the 2nd question that's all clear now too.

Cheers,

Imogen

**Reply by ISchwandner** [reply]
Got it working now. Many thanks!
